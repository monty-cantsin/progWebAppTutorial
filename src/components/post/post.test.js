import React from 'react'
import ReactDOM from 'react-dom'
import ReactTestRender from 'react-test-renderer'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'

import Post from '../post/post'

test("Post component's title should NOT render as a link", () => {
	const component = shallow(<Post title='My Post' />)
	const tree = toJson(component)
	expect(tree).toMatchSnapshot()
})

test("Post component's title should render as a link", () => {
	const component = shallow(<Post title='My Post' titleLink />)
	const tree = toJson(component)
	expect(tree).toMatchSnapshot()
})

