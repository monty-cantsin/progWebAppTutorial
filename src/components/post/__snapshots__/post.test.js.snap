// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component's title should NOT render as a link 1`] = `
<div>
  <h2
    className="title"
  >
    My Post
  </h2>
  <p
    className="content"
  />
</div>
`;

exports[`Post component's title should render as a link 1`] = `
<div>
  <h2
    className="title"
  >
    <Link
      activeClassName=""
      activeOnlyWhenExact={false}
      activeStyle={Object {}}
      className="link"
      isActive={[Function]}
      replace={false}
      style={Object {}}
      to="/post/undefined"
    >
      My Post
    </Link>
  </h2>
  <p
    className="content"
  />
</div>
`;
